daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -623,24 +623,18 @@ case class WholeExpression(
     }
 
     if (!allowCoercion) {
-      val (relevantExpr: Expression, detailMsg: String) =
-        Conversion.polymorphicExpressionDiagnostics(inherentType, targetType, subExpr)
       if (tunable.allowExpressionResultCoercion) {
         SDW(
           WarnID.DeprecatedExpressionResultCoercion,
-          "In expression %s, result type (%s) should be manually cast to the expected type (%s) with the appropriate constructor." +
-            "Performing deprecated automatic conversion.%s",
-          relevantExpr.text,
+          "Result type (%s) should be manually cast to the expected type (%s) with the appropriate constructor." +
+            "Performing deprecated automatic conversion.",
           inherentType,
-          targetType,
-          detailMsg)
+          targetType)
       } else {
         SDE(
-          "In expression %s, result type (%s) must be manually cast to the expected type (%s) with the approrpriate constructor.%s",
-          relevantExpr.text,
+          "Result type (%s) must be manually cast to the expected type (%s) with the approrpriate constructor.",
           inherentType,
-          targetType,
```

Comment:
Should we add back "in expression %s" for these error messages?

Reply:
Done